presto-main/src/main/java/com/facebook/presto/operator/scalar/QuantileDigestFunctions.java

```diff
@@ -101,6 +102,41 @@ public static Block valuesAtQuantilesBigint(@SqlType("qdigest(bigint)") Slice in
         return output.build();
     }
 
+    @ScalarFunction("scale_qdigest")
+    @Description("Scale a quantile digest according to a new weight")
+    @SqlType("qdigest(double)")
+    public static Slice scaleQuantileDigestDouble(@SqlType("qdigest(double)") Slice input, @SqlType(StandardTypes.DOUBLE) double scale)
+    {
+        return scaleQuantileDigest(input, scale);
+    }
+
+    @ScalarFunction("scale_qdigest")
+    @Description("Scale a quantile digest according to a new weight")
+    @SqlType("qdigest(real)")
+    public static Slice scaleQuantileDigestReal(@SqlType("qdigest(real)") Slice input, @SqlType(StandardTypes.DOUBLE) double scale)
+    {
+        return scaleQuantileDigest(input, scale);
+    }
+
+    @ScalarFunction("scale_qdigest")
+    @Description("Scale a quantile digest according to a new weight")
+    @SqlType("qdigest(bigint)")
+    public static Slice scaleQuantileDigestBigint(@SqlType("qdigest(bigint)") Slice input, @SqlType(StandardTypes.DOUBLE) double scale)
+    {
+        return scaleQuantileDigest(input, scale);
+    }
+
+    private static Slice scaleQuantileDigest(Slice input, double scale)
+    {
+        if (scale < 0) {
```

Comment:
Nit: let's make it `<= 0` (or even `< 1e-5` since that is the constant used in the data structure)